### PR TITLE
Add solr monitor

### PR DESCRIPTION
**Pull request by maxkadel**
Add a Solr health monitor.

- Pinning to sqlite3 ~> 1.3 until the fix for this issue is released - https://github.com/rails/rails/pull/51592 (should be in 7.1.4)
- Update gemfiles using Appraisal

[Example PR](https://github.com/pulibrary/orangelight/pull/4062) adding this to a Blacklight-based application using Solr

### CODE REVIEW

**Review by tpendragon — APPROVED** [review]
Hoping we can get this in, we've got this custom check in a bunch of our applications right now.

**Comment by lbeder** [comment]
This is amazing @maxkadel! I'll merge #122, so that the tests could pass, and then get back to this PR

**Comment by lbeder** [comment]
@maxkadel can you rebase now and we'll try the CI again?

File: README.md [hunk 1]
```diff
@@ -312,6 +313,10 @@ Please note that `url` or `connection` can't be used at the same time.
 
 * `queue_size`: the size (maximum) of a queue which is considered unhealthy (the default is 100).
 
+### Solr
+
+* `url`: the URL used to connect to your Solr instance - must be a string. You can also use `url` to explicitly configure authentication (e.g., `'http://user:pass@example.solr.com:8983/'`)
+
```
Comment (lbeder) [inline README.md:319]: Could you update the example to `https`? 😅

Comment (maxkadel) [reply README.md:319]: Done!

**Pull request merged**